src/components/ResourceBrowser/ResourceBrowser.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useCallback, useMemo } from 'react'
+import { NavLink, useHistory } from 'react-router-dom'
+import { showError, getUserRole, DevtronProgressing, useAsync } from '@devtron-labs/devtron-fe-common-lib'
+import PageHeader from '../common/header/PageHeader'
+import { sortObjectArrayAlphabetically } from '../common'
+import { ALL_NAMESPACE_OPTION, K8S_EMPTY_GROUP, SIDEBAR_KEYS } from './Constants'
+import { URLS } from '../../config'
+import ClusterSelectionList from '../ClusterNodes/ClusterSelectionList'
+import { getClusterList } from '../ClusterNodes/clusterNodes.service'
+import { ClusterDetail } from '../ClusterNodes/types'
+import { ReactComponent as Add } from '../../assets/icons/ic-add.svg'
+import './ResourceBrowser.scss'
+
+/* NOTE: anonymous funcitons are references
+ * when we export default we need to pass a reference or value
+ * Thus not naming function */
+const ResourceBrowser: React.FC<Record<string, never>> = () => {
+    const { push } = useHistory()
+
+    /* this list is to be used by ClusterSelectionList to refresh on demand;
+     * thus not immediately fetching results, it will only run when reloadClusterList is called */
+    const [clusterListLoading, clusterList, clusterListError, reloadClusterList] = useAsync(getClusterList, [], false)
+    const [initialLoading, data, error] = useAsync(() =>
+        Promise.all([getClusterList(), window._env_.K8S_CLIENT ? null : getUserRole()]),
+    )
+    /* transpose the data */
+    const [clusterListData, userRoleData] = data || []
+
+    const sortedClusterList: ClusterDetail[] = useMemo(
+        () =>
+            sortObjectArrayAlphabetically(
+                clusterList?.result || clusterListData?.result || [],
+                'name',
+            ) as ClusterDetail[],
+        [clusterListLoading, initialLoading],
+    )
+
+    const addClusterButton = useCallback(() => {
+        /* functions returning JSX should be memoized */
+        return (
+            <>
+                <NavLink
+                    className="flex dc__no-decor cta small h-28 pl-8 pr-10 pt-5 pb-5 lh-n fcb-5 mr-16"
+                    to={URLS.GLOBAL_CONFIG_CLUSTER}
+                >
+                    <Add
+                        data-testid="add_cluster_button"
+                        className="icon-dim-16 mr-4 fcb-5 dc__vertical-align-middle"
+                    />
+                    Add cluster
+                </NavLink>
+                <span className="dc__divider" />
+            </>
+        )
+    }, [])
+
+    const onChangeCluster = (selected): void => {
+        const url = `${URLS.RESOURCE_BROWSER}/${selected.value}/${ALL_NAMESPACE_OPTION.value}/${SIDEBAR_KEYS.nodeGVK.Kind.toLowerCase()}/${K8S_EMPTY_GROUP}`
+        push(url)
+    }
+
+    if (initialLoading) {
+        return (
+            <div style={{ height: 'calc(100vh - 48px)' }}>
+                <DevtronProgressing parentClasses="h-100 flex bcn-0" classes="icon-dim-80" />
```

Comment:
Are we sure we need this or <Progressing />

Reply:
The loading state is existing behaviour. But it should be conditionally rendered between clusterSelectionList and the loading state - i.e if initialLoading === true, show this loading state otherwise show cluster selection list. But the PageHeader should be shown irrespective of the loading state.